### PR TITLE
chore: update TM dep + go.mod cleanup

### PR DESCRIPTION
**Pull request by alexanderbez**
## Description

- Update TM to v0.34.19
- Cleanup go.mod
- Update go 1.17

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -25,48 +25,111 @@ require (
...
+	// Fix upstream GHSA-h395-qcrw-5vmq vulnerability.
+	// TODO Remove it: https://github.com/cosmos/cosmos-sdk/issues/10409
+	github.com/gin-gonic/gin => github.com/gin-gonic/gin v1.7.0
+	github.com/gogo/protobuf => github.com/regen-network/protobuf v1.3.3-alpha.regen.1
+	github.com/tendermint/tm-db => github.com/osmosis-labs/tm-db v0.6.5-0.20210911033928-ba9154613417
 
-replace github.com/cosmos/iavl => github.com/osmosis-labs/iavl v0.17.3-osmo-v7
+	// latest grpc doesn't work with with our modified proto compiler, so we need to enforce
+	// the following version across all dependencies.
+	google.golang.org/grpc => google.golang.org/grpc v1.33.2
```
Comment (mattverse) [inline go.mod:134]: Is there a reason we prefer grpc v1.33.2 instead of using  google.golang.org/grpc v1.45.0?

File: server/util.go [hunk 2]
```diff
@@ -271,12 +272,13 @@ func AddCommands(rootCmd *cobra.Command, defaultNodeHome string, appCreator type
...
 		VersionCmd(),
 	)
+
 	startCmd := StartCmd(appCreator, defaultNodeHome)
 	addStartFlags(startCmd)
 
 	rootCmd.AddCommand(
 		startCmd,
-		UnsafeResetAllCmd(),
+		tmcmd.ResetAllCmd,
```
Comment (mattverse) [inline server/util.go:284]: Just curious: did you ever get to test this cmd with the changes applied?

File: go.mod (hunk 1 above)
Comment (alexanderbez) [reply go.mod:134]: No, I'm not aware of any reason. In fact, the SDK `main` branch has `v1.45.0`. We can update in a subsequent PR 👍

File: server/util.go (hunk 2 above)
Comment (alexanderbez) [reply server/util.go:284]: I did! The command works, but panics lol. I think the TM is on it and will release a patch update soon.

Comment (alexanderbez) [reply server/util.go:284]: Actually, I'll fix it myself.

Comment (alexanderbez) [reply server/util.go:284]: NVM: Already fixed. https://github.com/tendermint/tendermint/pull/8258

I think we can just merge this and update later.

Comment (sunnya97) [reply server/util.go:284]: Can you create an issue to remember to update it

Comment (alexanderbez) [reply server/util.go:284]: Nvm, just gonna wait till they get a patch out as it shouldn't take long :p

Comment (ValarDragon) [reply server/util.go:284]: LGTM! I'm also happy for us to just use a commit off of the v0.34.x branch, and not be blocked on the next release being made unless its happening very soon

Comment (ValarDragon) [inline server/util.go:284]: Should we add the reset state command as well? (The new command that TM introduices)

Comment (alexanderbez) [reply server/util.go:284]: Yup, I'll add that.

Comment (alexanderbez) [reply server/util.go:284]: SGTM.

File: server/util.go [hunk 3]
```diff
@@ -279,7 +282,6 @@ func AddCommands(rootCmd *cobra.Command, defaultNodeHome string, appCreator type
 	rootCmd.AddCommand(
 		startCmd,
 		tmcmd.ResetAllCmd,
-		tendermintCmd,
```
Comment (ValarDragon) [inline server/util.go:280]: Was deleting this an accident? Not sure how the tm commands get in without this line

Comment (alexanderbez) [reply server/util.go:280]: No I moved it to the tendermint sub-command with all the other Tendermint proxied commands.

**Comment by alexanderbez** [comment]
This is blocked on another release which needs to include my fix: https://github.com/tendermint/tendermint/pull/8270

**Comment by alexanderbez** [comment]
@Mergifyio backport v0.45.0x-osmo-v8

**Pull request merged**